### PR TITLE
Resolve issues that break docker builds in #1154 of build repo

### LINKED ISSUE #1154: [Bug]: Docker build fails without security plugin
### Describe the bug

For 1.3.0 we don't have plugins (yet).

```
#25 [linux/arm64 linux_stage_0 5/8] RUN tar -xzpf /tmp/opensearch/opensearch-`uname -p`.tgz -C /usr/share/opensearch --strip-components=1 && rm -rf /tmp/opensearch &&     chmod 750 /usr/share/opensearch/plugins/opensearch-security/tools/* &&     mkdir -p /usr/share/opensearch/data && chown -R 1000:1000 /usr/share/opensearch/data
#25 CANCELED
------
 > [linux/amd64 linux_stage_0 5/8] RUN tar -xzpf /tmp/opensearch/opensearch-`uname -p`.tgz -C /usr/share/opensearch --strip-components=1 && rm -rf /tmp/opensearch &&     chmod 750 /usr/share/opensearch/plugins/opensearch-security/tools/* &&     mkdir -p /usr/share/opensearch/data && chown -R 1000:1000 /usr/share/opensearch/data:
#12 3.453 chmod: cannot access '/usr/share/opensearch/plugins/opensearch-security/tools/*': No such file or directory
------
opensearch.al2.dockerfile:34
--------------------
  33 |     COPY opensearch-*.tgz /tmp/opensearch/
  34 | >>> RUN tar -xzpf /tmp/opensearch/opensearch-`uname -p`.tgz -C $OPENSEARCH_HOME --strip-components=1 && rm -rf /tmp/opensearch && \
  35 | >>>     chmod 750 $OPENSEARCH_HOME/plugins/opensearch-security/tools/* && \
  36 | >>>     mkdir -p $OPENSEARCH_HOME/data && chown -R $UID:$GID $OPENSEARCH_HOME/data
  37 |     COPY opensearch-docker-entrypoint.sh opensearch-onetime-setup.sh $OPENSEARCH_HOME/
--------------------
error: failed to solve: process "/bin/sh -c tar -xzpf /tmp/opensearch/opensearch-`uname -p`.tgz -C $OPENSEARCH_HOME --strip-components=1 && rm -rf /tmp/opensearch &&     chmod 750 $OPENSEARCH_HOME/plugins/opensearch-security/tools/* &&     mkdir -p $OPENSEARCH_HOME/data && chown -R $UID:$GID $OPENSEARCH_HOME/data" did not complete successfully: exit code: 1
```

### To reproduce

Any CI build fails this way today.

### Expected behavior

Docker build should be skipped or produce some output just with OpenSearch core.

### Host / Environment

_No response_

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by peterzhuamazon**
Signed-off-by: Peter Zhu <zhujiaxi@amazon.com>

### Description
Resolve issues that break docker builds in #1154 of build repo

Initially #1154 stated that docker build is not running due to both security and PA plugin not present in 1.3 yet, thus cause `COPY` function to fail in dockerfiles of OpenSearch, as the target dir not exist due to no plugins installed.

Previously a PR #1162 tried to address the issue but ends up mix the usages between `COPY` and `cp`.

After some investigation, there are three issues that present in the codebase that prevents docker build on 1.3 branch running correctly:

1. #1145 change the S3 folder structure, cause opensearch/dashboards tarball not able to download correctly.
2. As stated above, `COPY` and `cp` means different context in dockerfile, one cannot use `cp` to replace `COPY`.
3. In #1133, a incorrect username is append for docker hub login, cause docker image to not be able to push to remote.

More logs in: https://github.com/opensearch-project/opensearch-build/issues/1154#issuecomment-981894695

This PR addresses all the issues above, ensure the image can be correctly built for both OpenSearch and OpenSearch-Dashboards. There are further issues to be address for docker run later.
 
### Issues Resolved
#1154
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peternied** [comment]
I am out of the loop on this change, could you update the issue to describe what is broken and how this change fixes it?

File: docker/release/dockerfiles/opensearch.al2.dockerfile [hunk 1]
```diff
@@ -27,20 +30,19 @@ RUN yum update -y && yum install -y tar gzip shadow-utils which && yum clean all
...
+# Copy artifacts and configurations to corresponding directories
+COPY * $TEMP_DIR
+RUN ls -l $TEMP_DIR && \
+    tar -xzpf /tmp/opensearch/opensearch-`uname -p`.tgz -C $OPENSEARCH_HOME --strip-components=1 && \
+    mkdir -p $OPENSEARCH_HOME/data && chown -Rv $UID:$GID $OPENSEARCH_HOME/data && \
+    if [[ -d $SECURITY_PLUGIN_DIR ]] ; then chmod -v 750 $SECURITY_PLUGIN_DIR/tools/* ; fi && \
+    if [[ -d $PERFORMANCE_ANALYZER_PLUGIN_DIR ]] ; then cp -v $TEMP_DIR/performance-analyzer.properties $PERFORMANCE_ANALYZER_PLUGIN_DIR/pa_config/; fi && \
+    cp -v $TEMP_DIR/opensearch-docker-entrypoint.sh $TEMP_DIR/opensearch-onetime-setup.sh $OPENSEARCH_HOME/ && \
+    cp -v $TEMP_DIR/log4j2.properties $TEMP_DIR/opensearch.yml $OPENSEARCH_HOME/config/ && \
+    ls -l $OPENSEARCH_HOME && rm -rf $TEMP_DIR
```
Comment (dblock) [inline docker/release/dockerfiles/opensearch.al2.dockerfile:45]: minor: put `rm -rf` on a new line so we can see it better

**Review by dblock — APPROVED** [review]

**Comment by peterzhuamazon** [comment]
> I am out of the loop on this change, could you update the issue to describe what is broken and how this change fixes it?

https://github.com/opensearch-project/opensearch-build/issues/1154#issuecomment-981894695

File: docker/release/dockerfiles/opensearch.al2.dockerfile (hunk 1 above)
Comment (peterzhuamazon) [reply docker/release/dockerfiles/opensearch.al2.dockerfile]: Resolved.

**Comment by peternied** [comment]
> > I am out of the loop on this change, could you update the issue to describe what is broken and how this change fixes it?
> 
> [#1154 (comment)](https://github.com/opensearch-project/opensearch-build/issues/1154#issuecomment-981894695)

There is a lot of content inside that comment, let alone issue, and I think it would be worthwhile to attach an explanation to this pull request.

**Review by tianleh — APPROVED** [review]

**Comment by peterzhuamazon** [comment]
> > > I am out of the loop on this change, could you update the issue to describe what is broken and how this change fixes it?
> > 
> > 
> > [#1154 (comment)](https://github.com/opensearch-project/opensearch-build/issues/1154#issuecomment-981894695)
> 
> There is a lot of content inside that comment, let alone issue, and I think it would be worthwhile to attach an explanation to this pull request.

Add information in the PR description now.
Thanks.

**Pull request merged**